examples/pytorch/nlp/huggingface_models/text-classification/pruning/hpo/run_glue_no_trainer.py

```diff
@@ -0,0 +1,669 @@
+            accelerator.wait_for_everyone()
+            unwrapped_model = accelerator.unwrap_model(model)
+            unwrapped_model.save_pretrained(args.output_dir, save_function=accelerator.save)
+            if accelerator.is_main_process:
+                tokenizer.save_pretrained(args.output_dir)
+                repo.push_to_hub(
+                    commit_message=f"Training in progress epoch {epoch}", blocking=False, auto_lfs_prune=True
+                )
+        if args.output_dir is not None:
+            accelerator.wait_for_everyone()
+            unwrapped_model = accelerator.unwrap_model(model)
+            file = os.path.join(args.output_dir, f"epoch{epoch}")
+            unwrapped_model.save_pretrained(file)
+            # unwrapped_model.save_pretrained(args.output_dir, save_function=accelerator.save)
+            if accelerator.is_main_process:
+                tokenizer.save_pretrained(args.output_dir)
+                # if args.push_to_hub:
+                #     repo.push_to_hub(commit_message="End of training", auto_lfs_prune=True)
+
+    pattern_sparsity_over_conv_linear, element_sparsity_over_conv_linear, element_sparsity_over_all = pruner.get_sparsity_ratio()
+    print(pattern_sparsity_over_conv_linear, element_sparsity_over_conv_linear, element_sparsity_over_all)
+    if args.output_dir is not None:
+        accelerator.wait_for_everyone()
+        unwrapped_model = accelerator.unwrap_model(model)
+        file = os.path.join(args.output_dir, f"epoch{epoch}.pytorch.bin")
+        unwrapped_model.save_pretrained(file)
+        # unwrapped_model.save_pretrained(args.output_dir, save_function=accelerator.save)
+        if accelerator.is_main_process:
+            tokenizer.save_pretrained(args.output_dir)
+            if args.push_to_hub:
+                repo.push_to_hub(commit_message="End of training", auto_lfs_prune=True)
+
+    if args.task_name == "mnli":
+        # Final evaluation on mismatched validation set
+        eval_dataset = processed_datasets["validation_mismatched"]
+        eval_dataloader = DataLoader(
+            eval_dataset, collate_fn=data_collator, batch_size=args.per_device_eval_batch_size
+        )
+        eval_dataloader = accelerator.prepare(eval_dataloader)
+
+        model.eval()
+        for step, batch in enumerate(eval_dataloader):
+            outputs = model(**batch)
+            predictions = outputs.logits.argmax(dim=-1)
+            metric.add_batch(
+                predictions=accelerator.gather(predictions),
+                references=accelerator.gather(batch["labels"]),
+            )
+
+        eval_metric = metric.compute()
+        logger.info(f"mnli-mm: {eval_metric}")
+    return eval_metric
+
+
+if __name__ == "__main__":
+#     main()
+    os.environ['CUDA_VISIBLE_DEVICES'] = '1'
+    import time
+    from neural_compressor.compression.hpo import XgbSearcher, DiscreteSearchSpace, ContinuousSearchSpace
```

Comment:
is it better to  combine DiscreteSearchSpace and ContinuousSearchSpace ?

Reply:
the style of inc is to wrap all the stuff in one API, please hold on this pr, we need to carefully design the interface

Reply:
done